18F/identity-idp
# LG-10760: refactor  validations for DocPiiForm

## 🎫 Ticket
[LG-10760](https://cm-jira.usa.gov/browse/LG-10760)

## 🛠 Summary of changes
Refactor DocPiiForm to user ActiveModel::Validation helpers that surfaces _all_ validation errors.

## 📜 Testing Plan

Provide a checklist of steps to confirm the changes.

1. upload document  with missing dob
2. upload document with dob too young (ie: 2023/10/31)
3. upload document with missing address
4. upload document with invalid zipcode (too short or not a string)
5. upload document with invalid jurisdiction (ie: 'UK')
6. upload document with invalid state (less than or more than 2 characters)


## 👀 Screenshots
<img width="635" alt="Screenshot 2023-10-30 at 5 48 32 PM" src="https://github.com/18F/identity-idp/assets/1261794/4b67fbfe-1def-4565-8700-f4db921c7702">
<img width="625" alt="Screenshot 2023-10-30 at 5 48 09 PM" src="https://github.com/18F/identity-idp/assets/1261794/ade6b20c-3216-4d36-b430-e289f79e80d8">
<img width="640" alt="Screenshot 2023-10-30 at 5 47 40 PM" src="https://github.com/18F/identity-idp/assets/1261794/98397b00-0ebb-4f60-a558-f5721ac6da6a">
<img width="638" alt="Screenshot 2023-10-30 at 5 53 57 PM" src="https://github.com/18F/identity-idp/assets/1261794/beba86c9-3295-4242-95a6-cc03b24fa377">
<img width="630" alt="Screenshot 2023-10-30 at 5 53 31 PM" src="https://github.com/18F/identity-idp/assets/1261794/7e5d4d7c-e4d7-4aff-a574-f00610326461">
